File: src/main/java/com/dh/ondot/schedule/api/response/AlarmSwitchResponse.java
package com.dh.ondot.schedule.api.response;

import com.dh.ondot.core.util.TimeUtils;
import com.dh.ondot.schedule.domain.Schedule;

import java.time.LocalDateTime;

public record AlarmSwitchResponse(
        Long scheduleId,
        Boolean isEnabled,
        LocalDateTime updatedAt
) {
    public static AlarmSwitchResponse from(Schedule schedule) {
        return new AlarmSwitchResponse(
                schedule.getId(),
                schedule.getDepartureAlarm().isEnabled(),
                TimeUtils.toSeoulDateTime(schedule.getUpdatedAt())
        );
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/presentation/response/AlarmSwitchResponse.kt
package com.dh.ondot.schedule.presentation.response

import com.dh.ondot.core.util.TimeUtils
import com.dh.ondot.schedule.domain.Schedule
import java.time.LocalDateTime

data class AlarmSwitchResponse(
    val scheduleId: Long?,
    val isEnabled: Boolean,
    val updatedAt: LocalDateTime?,
) {
    companion object {
        fun from(schedule: Schedule): AlarmSwitchResponse {
            return AlarmSwitchResponse(
                scheduleId = schedule.id,
                isEnabled = schedule.departureAlarm!!.isEnabled,
                updatedAt = TimeUtils.toSeoulDateTime(schedule.updatedAt),
            )
        }
    }
}
